### PR TITLE
[IndexTable.Row] Fix `tone` interaction states; remove `subdued` and `status`

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

This PR removes the deprecated `status` and `subdued` props. It also fixes the missing and buggy interaction states when `IndexTable.Row` has a `tone`.

| Before | After |
|--------|--------|
|<video src="https://github.com/Shopify/polaris/assets/18447883/2f2e2f76-adbc-4bed-a18b-c09a23b7542e"/>|<video src="https://github.com/Shopify/polaris/assets/18447883/4b097d18-4c0c-4fec-b561-9fc9ba2c55b9"/>|

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 1]
```diff
@@ -825,60 +858,6 @@ $loading-panel-height: 53px;
       z-index: auto;
     }
   }
-
-  // stylelint-disable selector-max-class, selector-max-combinators, selector-max-specificity -- generated by polaris-migrator DO NOT COPY
```
Comment (chloerice) [inline polaris-react/src/components/IndexTable/IndexTable.scss:829]: Styling the sticky cell down here causes unwanted specificity that requires unnecessary redundancy. Added :last-child to the decls for the various interaction states.

File: polaris.shopify.com/content/version-guides/migrating-from-v11-to-v12.md [hunk 2]
```diff
@@ -31,10 +31,24 @@ order: 1
...
 
+- Replace `status` with `tone`
+
 `npx @shopify/polaris-migrator react-rename-component-prop <path> --componentName="Badge" --from="status" --to="tone"`
 
+- Replace `statusAndProgressLabelOverride` with `toneAndProgressLabelOverride`
+
 `npx @shopify/polaris-migrator react-rename-component-prop <path> --componentName="Badge" --from="statusAndProgressLabelOverride" --to="toneAndProgressLabelOverride"`
 
+**IndexTable.Row**
```
Comment (jesstelford) [inline polaris.shopify.com/content/version-guides/migrating-from-v11-to-v12.md]: I think this should be a h3?

```suggestion
### IndexTable.Row
```

File: polaris-react/src/components/IndexTable/components/Row/tests/Row.test.tsx [hunk 3]
```diff
@@ -430,37 +430,61 @@ describe('<Row />', () => {
...
       </Row>,
     );
 
     expect(row).toContainReactComponent('tr', {
-      className: 'TableRow statusSubdued',
+      className: 'TableRow toneSuccess',
+    });
+  });
+
+  it('applies warning tone styles when tone prop is set to "success"', () => {
```
Comment (jesstelford) [inline polaris-react/src/components/IndexTable/components/Row/tests/Row.test.tsx]:
```suggestion
  it('applies warning tone styles when tone prop is set to "warning"', () => {
```

File: polaris-react/src/components/IndexTable/components/Row/tests/Row.test.tsx [hunk 4]
```diff
@@ -430,37 +430,61 @@ describe('<Row />', () => {
...
+        <td />
+      </Row>,
+    );
+
+    expect(row).toContainReactComponent('tr', {
+      className: 'TableRow toneWarning',
+    });
+  });
+
+  it('applies critical tone styles when tone prop is set to "success"', () => {
```
Comment (jesstelford) [inline polaris-react/src/components/IndexTable/components/Row/tests/Row.test.tsx]:
```suggestion
  it('applies critical tone styles when tone prop is set to "critical"', () => {
```

**Review by sophschneider — APPROVED** [review]

**Pull request merged**